.github/workflows/verify.yml

```diff
@@ -13,65 +13,75 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: Go
-# name: verify TODO rename (requires repo settings update)
-
+name: Check Unit and Integration tests
 # yamllint disable-line rule:truthy
 on:
   push:
-    branches: [master]
+    branches:
+      - master
   pull_request:
-    branches: [master, issue-99-cosmos-upgrade-v0.44]
+    branches:
+      - master
+      - issue-99-cosmos-upgrade-v0.44
   workflow_dispatch:
 
 jobs:
-
-  build:
-    name: Build  # TODO remove (requires repo settings update)
+  changes:
     runs-on: ubuntu-latest
+    outputs:
+      go: ${{ steps.filter.outputs.go }}
+      src: ${{ steps.filter.outputs.src }}
     steps:
-
-      - name: Build
-        run: true
-
-  unit:
-    name: Unit tests  # TODO remove (requires repo settings update)
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        if: ${{ !env.ACT }}
+        id: filter
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          filters: |
+            go:
+              - '**/*.go'
+              - 'go.mod'
+              - 'go.sum'
+            src:
+              - '!**.md'
+              - '!.github/*'
+  unit-tests:
+    name: Check unit tests
     runs-on: ubuntu-latest
+    needs:
+      - lint
     steps:
-
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v2
+      - uses: actions/setup-go@v2
         with:
           go-version: ^1.16
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
       - uses: actions/cache@v2
+        if: ${{ !env.ACT }}
         with:
           path: |
             ~/.cache/go-build
             ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
-
-      - name: Run units
-        run: make test
-
-  integration:
-    name: Integration tests  # remove (requires repo settings update)
+      - uses: haveyoudebuggedit/gotestfmt-action@v2
+      - name: Run tests
+        run: |
+          set -euo pipefail
+          # shellcheck disable=SC2046
+          go test -json -v $(go list ./... | grep -v '/integration_tests') 2>&1 | tee /tmp/gotest.log | gotestfmt
+  integration-tests:
+    if: needs.changes.outputs.src == 'true'
```

Comment:
I believe we need to run integration tests on any changes in .go files, as well as on any changes in the integration test source code (`integration_tests` folder).

Reply:
Technically, the regex in `src` covers any change to code (go, typescript, bash, etc.) except in documentation or GitHub actions

Reply:
BTW the `src` filter rule didn't work correctly, as OR is used in case of multiple regexps. So, it was true in most of the cases (when for example we had changes in .md files only).
It's fixed in https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/300.